pkg/gardenlet/apis/config/v1alpha1/types.go

```diff
@@ -406,6 +406,14 @@ type GardenLoki struct {
 	Priority *int `json:"priority,omitempty" yaml:"priority,omitempty"`
 }
 
+// ShootNodeLogging contains configuration for the shoot node logging.
+type ShootNodeLogging struct {
+	// ShootPurposes denote which shoots can have node loggin by they purpose
+	ShootPurposes []string `json:"shootPurposes,omitempty" yaml:"shootPurposes,omitempty"`
+	// AllowAllExecptTesting allows all shoot of all purpose, except testing to have shoot node logging
+	AllowAllExecptTesting bool `json:"allowAllExecptTesting,omitempty" yaml:"allowAllExecptTesting,omitempty"`
```

Comment:
If this is optional, add the `// +optional` tag and make `*bool`. If it's not optional remove the `omitempty` tags.

Reply:
I have removed this option.